File: codeforces/1257/C.py
for _ in range(int(input())):
    n = int(input())
    arr = list(map(int, input().split()))
    mem = {}

    for i in range(n):
        try:
            mem[arr[i]].append(i)
        except:
            mem[arr[i]] = [i]

    sub = list(mem.values())
    ans = 10**7
    for i in sub:
        l = len(i)
        if l>1:
            for j in range(1,l):
                if (i[j]-i[j-1])<ans:
                    ans = (i[j]-i[j-1])
    if ans==10**7:
        print(-1)
    else:
        print(ans+1)
            
    

File: codeforces/1257/C.cpp
#include <bits/stdc++.h>
using namespace std;
#define ll long long
#define fast ios_base::sync_with_stdio(false);cin.tie(NULL);

int main(){
    fast;
    ll t;
    cin>>t;
    while(t--){
        ll n;
        cin>>n;
        ll arr[n+1];
        ll mem[n+1] = {0};
        ll ans = 10000000;
        for(ll i=1;i<=n;i++){
            cin>>arr[i];
            if(mem[arr[i]]!=0){
                ans = min(ans, i-mem[arr[i]]);
                mem[arr[i]] = i;
            }
            else{
                mem[arr[i]]= i;
            }
        }
        if(ans==10000000){
            cout<<-1<<"\n";
        } else
            cout<<ans+1<<"\n";
    }
    return 0;
}
